### DISCUSSION TITLE
Color quantization on ImageMagick vs Indexing colors on Photoshop

### DISCUSSION THREAD

**Discussion by friaca** [ImageMagick/ImageMagick#2365] [Help] [answered]
I want to reduce the number of colors of an image, I tried several IM commands, the one that got the closest expected result was `convert -dither None -colors 5 image.jpg PNG8:image_reduced.png` but this "removes" the red in favour of shades of blue. 
I tried doing it by changing the image mode on Photoshop to "Indexed colors" and it gave me the expected result. Is there a way I can replicate the Photoshop output with IM?

[Source image](https://imgur.com/qO8rE7E)
[ImageMagick output](https://imgur.com/276u46m)
[Photoshop output](https://imgur.com/aXYsMLy)

**Comment by fmw42** [comment, marked as answer]
try -kmeans. See https://imagemagick.org/discourse-server/viewtopic.php?f=4&t=37301

**Reply by friaca** [reply]
Ohhhhh, this is exactly what I needed. Thank you so much! :)

**Reply by sergeevabc** [reply]
“The requested topic does not exist”, but Wayback Machine has a [copy](https://web.archive.org/web/20200805054742if_/https://imagemagick.org/discourse-server/viewtopic.php?f=4&t=37301).
